### PR TITLE
Adaptor Update for Unity Ads SDK 3.3.0 - New Banner API

### PR DESCRIPTION
**Pull request by fritzhuie**
Updated Adaptors to use the new SDK 3.3 banner API

The SDK 3.3 banner API changes should bring Unity Ads banner behavior in line with AdMob's to simplify adaptor configurations.

Now takes in a banner size and supports multiple banner views

### CODE REVIEW

**Comment by jindoujiao** [comment]
@googlebot I consent

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    [strongConnector adapter:strongAdapter didReceiveAdView:bannerView];
+  }
+}
+
+
+- (void)bannerViewDidClick:(UADSBannerView *)bannerView {
+  id<GADMAdNetworkConnector> strongConnector = _connector;
+  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
+  if (strongAdapter && strongConnector) {
+    [strongConnector adapterDidDismissFullScreenModal:strongAdapter];
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Not mapped correctly. Should map to a click event, not a screen dismissal. Use 
`[strongConnector adapterDidGetAdClick:strongAdapter];` instead.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+  if (!_placementID){
+    NSString *description = [NSString
+                             stringWithFormat:@"Tried to initialize %@ with nil placement.", NSStringFromClass([UADSBannerView class])];
+    NSError *error = GADUnityErrorWithDescription(description);
+    [strongConnector adapter:strongAdapter didFailAd:error];
+    return;
+  }
+
+  adSize.size.width = adSize.size.width < 320 ? 320 : adSize.size.width;
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: If size coming in is less than 320x50, Unity should not be returning a larger ad. It should fail the request.

Also, ideally, UnityAds SDK should be able to fail the load if the ad size is less than 320x50, and adapter should be able to pass through their SDK's error. What happens if Unity gets a small size?

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  if (error) {
+    [strongConnector adapter:strongAdapter didFailAd:error];
+    return;
+  }
+
+  _placementID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityPlacementID] copy];
+
+  if (!_placementID){
+    NSString *description = [NSString
+                             stringWithFormat:@"Tried to initialize %@ with nil placement.", NSStringFromClass([UADSBannerView class])];
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Would be easier to just say "Unity banner" in these logs. Is this class name really going to change and/or be useful?

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  if (!strongConnector || !strongAdapter) {
+    NSLog(@"Adaptor Error: No GADMAdNetworkConnector or GADMAdNetworkAdapter found for getBannerWithSize:");
+    return;
+  }
+
+  NSError *error = nil;
+  if (error) {
+    [strongConnector adapter:strongAdapter didFailAd:error];
+    return;
+  }
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: We can remove this code.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+  _placementID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityPlacementID] copy];
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: `[strongConnector.credentials[kGADMAdapterUnityPlacementID] copy];`.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+- (void)bannerViewDidError:(UADSBannerView *)bannerView error:(UADSBannerError *)error {
+  id<GADMAdNetworkConnector> strongConnector = _connector;
+  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
+  if (strongConnector && strongAdapter) {
+    if ( error.code == UADSBannerErrorCodeNoFillError) {
+      [strongConnector adapter:strongAdapter didFailAd:GADUnityErrorWithDescription(@"Unity Ads banner failed to load - No Fill.")];
+    } else {
+      [strongConnector adapter:strongAdapter didFailAd:GADUnityErrorWithDescription(@"Unity Ads banner failed to load - Error.")];
+    }
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Better to make if statement only have the string, and to have 1 line of code to invoke the callback using the string. Less repetitive code.

**Comment by maninomula** [comment]
@fritzhuie , thanks for the PR. We have left few comments. Please resolve them so that we can move forward in releasing the adapter. Thanks.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 1 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Fixed it in the new commit.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 2 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: We will do no fill for the size smaller than 320*50. So remove the re-size here.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 3 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Remove the class name and use "Unity Banner" instead.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 4 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Removed.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 6 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Fixed it.

**Comment by jindoujiao** [comment]
@maninomula, thanks for the detailed comments.  please check out the latest commit which addressed most of them.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 5 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: fixed.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 7]
```diff
@@ -58,30 +58,21 @@ - (void)getBannerWithSize:(GADAdSize)adSize {
...
 
-  adSize.size.width = adSize.size.width < 320 ? 320 : adSize.size.width;
-  adSize.size.height = adSize.size.height < 50 ? 50 : adSize.size.height;
-
   _bannerAd = [[UADSBannerView alloc] initWithPlacementId:_placementID size:adSize.size];
 
   if (!_bannerAd) {
     NSString *description = [NSString
-                             stringWithFormat:@"%@ failed to initialize.", NSStringFromClass([UADSBannerView class])];
+                             stringWithFormat:@"%@ failed to initialize.", @"Unity banner"];
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: `[NSString stringWithFormat:@"Unity banner failed to initialize."];`

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m [hunk 8]
```diff
@@ -58,30 +58,21 @@ - (void)getBannerWithSize:(GADAdSize)adSize {
...
-    return;
-  }
-
-  _placementID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityPlacementID] copy];
+  _placementID = [strongConnector.credentials[kGADMAdapterUnityPlacementID] copy];
 
   if (!_placementID){
     NSString *description = [NSString
-                             stringWithFormat:@"Tried to initialize %@ with nil placement.", NSStringFromClass([UADSBannerView class])];
+                             stringWithFormat:@"Tried to initialize %@ with nil placement.", @"Unity banner"];
```
Comment (maninomula) [inline adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: `[NSString stringWithFormat:@"Tried to initialize Unity banner with nil placement."];`

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 3 above)
Comment (maninomula) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Use the string directly., no need of format here.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 7 above)
Comment (maninomula) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Ignore the previous comment. Use the string directly., no need of format here.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 8 above)
Comment (maninomula) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Ignore the previous comment. Use the string directly., no need of format here.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 7 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Removed the format, and use string directly.

File: adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m (hunk 8 above)
Comment (jindoujiao) [reply adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m]: Removed string format and use string directly. @maninomula

**Pull request merged**